hpat/hiframes/pd_series_ext.py

```diff
@@ -1270,23 +1270,18 @@ def typer(val1, val2):
 @overload(pd.Series)
 def pd_series_overload(data=None, index=None, dtype=None, name=None, copy=False, fastpath=False):
 
-    if index is not None:
-        def hpat_pandas_series_index_ctor_impl(
-                data=None,
-                index=None,
-                dtype=None,
-                name=None,
-                copy=False,
-                fastpath=False):
-            return hpat.hiframes.api.init_series(
-                hpat.hiframes.api.fix_df_array(data),
-                hpat.hiframes.api.fix_df_array(index),
-                name)
-
-        return hpat_pandas_series_index_ctor_impl
+    is_index_none = True if (isinstance(index, types.NoneType) or index is None) else False
 
     def hpat_pandas_series_ctor_impl(data=None, index=None, dtype=None, name=None, copy=False, fastpath=False):
-        return hpat.hiframes.api.init_series(hpat.hiframes.api.fix_df_array(data), index, name)
+
+        # use binop here as otherwise Numba's dead branch pruning doesn't work
```

Comment:
Please form it as regular doc string

Reply:
Docstrings are for auto generation of documentation. To provide description of classes/functions. Not for code explaining. But comments for code explaining

Reply:
@shssf Sorry, I don't fully understand what change you are requesting. Should I put a comment with TODO about one specific line of code in the function doc string? Or do you want to add general doc string for this function, not related to this comment?

Reply:
@kozlov-alexey I just thought that it is better to have
```
'''
comment
'''
```
instead
`# comment`